core/modules/user/user.module

```diff
@@ -10,6 +10,26 @@
 define('USERNAME_MAX_LENGTH', 60);
 
 /**
+ * Maximum length of user email text field.
+ */
+define('USERMAIL_MAX_LENGTH', 254);
```

Comment:
I think this constant is probably unnecessary. Let's just use the existing `EMAIL_MAX_LENGTH` which is already the same value of 254